pkg/kubelet/sharedcpus/sharedcpus.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sharedcpus
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	sharedCpusFileName     = "/etc/kubernetes/openshift-shared-cpus"
+	sharedCpusResourceName = "openshift.io/enabled-shared-cpus"
```

Comment:
We are using workload.openshift.io in some spots. Should this be namespaced to `workload.openshift.io/enabled-shared-cpus`?

Reply:
Is the `workload` namespace stands for features that are coupled with workload partition?
Because this feature can be activated regardless of workload partition state

Reply:
I'll defer to @haircommander and @mrunalp; they will know best.

Reply:
Yes, makes sense to put this under workload.openshift.io.

Reply:
The `workload.openshift.io/enabled-shared-cpus` is user-facing value (and we have others as well like the `OPENSHIFT_ISOLATED_CPUS` and  `OPENSHIFT_SHARED_CPUS`) - and it should be used in some other code areas such the admission plugin and CRIO. 
Instead of scatter those  user-facing values  I suggest to place them in central package and vendor from there.
https://github.com/openshift-kni/mixed-cpu-node-plugin/api might be a good fit IMHO
@mrunalp @rphillips WDYT?